packages/loader/container-loader/src/container.ts

```diff
@@ -1036,6 +1044,13 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             args.mode = "write";
         }
 
+        if (this.connectionState === ConnectionState.Disconnected) {
```

Comment:
I do not think this check is correct one. Client might be disconnected but trying to connect. I.e., if you call this API 10 times in a row in a tight loop, 10 events will be fired. I do not think that's what clients want.

The problem with events like that is that you will likely need to expose (eventually) all the details of internal machinery. And internals will change over time (for example, we used to not drop connection immediately on setAutoReconnect(false), making it less clear what events like that actually track / mean).

I'd rather say - clients can track this kind of timing only for very small set of scenarios, like boot, where they have full control over sequencing (when they call various APIs).